docs/architecture/ADR-004-IBC-whitelist.md

```diff
@@ -0,0 +1,29 @@
+# ADR 004: IBC Whitelist/Allowlist
+<!-- disable markdownlint MD010 because Go code snippet uses tabs -->
+<!-- markdownlint-disable MD010 -->
+
+## Terminology
+
+All terminolofy is defined in [ICS 024](https://github.com/cosmos/ibc/tree/main/spec/core/ics-024-host-requirements) and some references are from correspodnign implementation of [ICS 002](https://github.com/cosmos/ibc/tree/main/spec/core/ics-002-client-semantics) in [ibc-go](https://github.com/cosmos/ibc-go/blob/da1b7e0aaf4b7d466b1a7d1ed4f5d81149ff1d5b/modules/core/02-client)
+
+## Changelog
+
+- 2022-03-03: Initial Commit
+
+## Context
+
+While enabling IBC, we want to connect, and allow messages from selected chains. However, we dont want the ability for an arbitrary entity to create an IBC connection with a zone. This is so that we can keep the state machine as minimal and focused as possible.
```

Comment:
[typo]

```suggestion
While enabling IBC, we want to connect, and allow messages from selected chains. However, we don't want the ability for an arbitrary entity to create an IBC connection with a zone. This is so that we can keep the state machine as minimal and focused as possible.
```